nix/machines/default.nix

```diff
@@ -0,0 +1,68 @@
+{
+  self,
+  lib,
+  ...
+}:
+let
+  allEnvironmentPaths = lib.filter (path: lib.hasPrefix "example-setup-" (builtins.baseNameOf path)) (
+    lib.attrValues (builtins.readDir ../test-environments)
+  );
+
+  systems = [
+    "x86_64-linux"
+    "aarch64-linux"
+    "x86_64-darwin"
+    "aarch64-darwin"
+  ];
+  allEnvironmentMachines = lib.foldl' (
+    acc: path:
+    let
+      baseName = builtins.baseNameOf path;
+      machinesForPath = lib.listToAttrs (
+        lib.map (system: {
+          name = if system == "x86_64-linux" then baseName else "${baseName}-${system}";
+          value = testEnvironmentConfig path system;
+        }) systems
+      );
+    in
+    acc // machinesForPath
+  ) { } allEnvironmentPaths;
```

Comment:
Why the preference for `x86_64-linux`? Is it not better to make them the same?

Reply:
Agreed, though if we test solely `x86_64-linux`, then perhaps the extra complexity is not worth it.